From 0bba686123af9b7af05236b5f1c8fca3d56b828e Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Wed, 23 Mar 2022 09:18:23 +0000
Subject: [PATCH] Add -fno-plt to default build flags

As we default to BIND_NOW and symbolic-functions we already skip
the procedural lookup table already, the PLT mostly sits useless,
removing it will marginally decrease package sizes and improve start
up times.

A few packages that employ lazy-linking (xorg) or programs doing
funky things (e.g. valgrind) will need the PLT so we'll add an opt-in
to ypkg for those packages.
---
 pisi/configfile.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index f569df8..f0c7cec 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -86,8 +86,8 @@ class BuildDefaults:
     jobs = "auto"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
-    cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
+    cflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fno-plt -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
+    cxxflags = "-mtune=generic -march=x86-64 -g2 -O2 -pipe -fPIC -fno-plt -D_FORTIFY_SOURCE=2 -fstack-protector-strong --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-vectorize -feliminate-unused-debug-types -Wall -Wno-error -Wp,-D_REENTRANT"
     ldflags = "-Wl,--copy-dt-needed-entries -Wl,-O1 -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000 -Wl,-Bsymbolic-functions -Wl,--sort-common"
     buildhelper = "ccache"
     compressionlevel = 1
